### DISCUSSION TITLE
Derating and temperature violation cost functions

### DISCUSSION THREAD

**Discussion by JBRDLR** [DLR-VSDC/IEEE-MVC-2023#31] [Q&A]
### Discussed in https://github.com/DLR-VSDC/IEEE-MVC-2023/discussions/22

<div type='discussions-op-text'>

<sup>Originally posted by **mariagrazia-tristano** February  1, 2023</sup>
Hello,

While looking at scores from some preliminary simulations I came across something odd. Looking at the baseline scores file, for some of the tracks the velocity derating cost function J_v and the temperature violation cost function J_T are zero, which is perfectly plausible when no derating is necessary and no temperature violation occurs. On the other hand, the two corresponding competitor cost functions can be bigger than 0, in case of temporary violation of the battery temperature constraint or in case derating is deemed necessary in some time instants.

However, looking at how the normalisation is performed, the score-checking MATLAB function seems to "punish" the competitor by setting the normalised cost to be infinite when the baseline cost is zero and the competitor one is bigger than zero. So if the competitor incurs in that second "if" case of the score-checking function, essentially the entire cost function becomes infinite. Is there a reason why it was set that way?

Moreover, since the temperature violation cost function in the Simulink file saves the maximum temperature without any following integration, wouldn't this imply that say a 1°C temperature violation for 1s would affect the cost function in the same way as if one kept the battery temperature constantly at 1°C above the maximum for the whole duration of the simulation?

Many thanks.</div>

**Comment by imeb** [comment]
Dear @mariagrazia-tristano, 
Thanks for your question! Please find below the reasoning:

We have considered the temperature constraint as a sensitive cost function in the sense that T_max is chosen as the critical temperature above which the battery will be damaged irreversibly. Hence, if there are tracks that the baseline policy is able to fulfill even with the cost of derating the velocity to not violate the temperature constraint (even instantaneously), it is more desirable.

For this reason, we also only look at the maximum temperature reached, and not the timespan at which it has been violated.

It is advised to develop your EMA such that to enhance the baseline for those tracks where the temperature is being violated or the derating is active while the temperature is still below the maximum threshold.


Thanks again for raising this question, and please feel free to reply to this message if you have further doubts.
Regards.

**Comment by mariagrazia-tristano** [comment]
Thank you for your reply. Two main points:

- The full paper in chapter 4B says "This set of constraints can be temporarily violated; however, these violation increases the risk of failure of this component and are penalized in the EMA performance score." so does this mean that for some tracks, specifically the ones for which the baseline returns J_T equal to 0, the temperature constraint cannot be violated?
- "if there are tracks that the baseline policy is able to fulfill even with the cost of derating the velocity to not violate the temperature constraint (even instantaneously), it is more desirable" however as I pointed out in my original question, the derating cost function suffers from the same problem as the temperature one: in the tracks where the baseline performance has alpha_v = 1 at all times (hence J_v = 0), if the competitor performance has at any point alpha_v less than 1 this will cause the integral in the simulink model to be bigger than zero, which leads to infinite J_v and infinite cost function. So is the derating "not allowed" as well for those tracks?

Many thanks.

**Comment by imeb** [comment]
Dear @mariagrazia-tristano,

> * This set of constraints can be temporarily violated

This rule always applies. You can violate any of the constraints, but there will be a penalty (possibly infinite in final score) in final score.

> * so does this mean that for some tracks, specifically the ones for which the baseline returns J_T equal to 0, the temperature constraint cannot be violated?
We expect competitors to improve the baseline EMA. For example, if the baseline metric J_x is zero, then your EMA should also be zero. (J_x, can refer to J_v, J_TC, etc.)


> * if the competitor performance has at any point alpha_v less than 1 this will cause the integral in the simulink model to be bigger than zero, which leads to infinite J_v and infinite cost function. So is the derating "not allowed" as well for those tracks?

The answer is yes, with the same reasoning as above.
Good luck!
